src/migrations/20181203153000_move_out_bundles_metadata.js

```diff
@@ -0,0 +1,38 @@
+/*
+Copyright: Ambrosus Technologies GmbH
+Email: tech@ambrosus.com
+
+This Source Code Form is subject to the terms of the Mozilla Public License, v. 2.0. If a copy of the MPL was not distributed with this file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+This Source Code Form is “Incompatible With Secondary Licenses”, as defined by the Mozilla Public License, v. 2.0.
+*/
+
+// eslint-disable-next-line import/prefer-default-export
+export const up = async (db, config, logger) => {
+  await db.collection('bundles').ensureIndex({bundleId : 1}, {unique: true});
+  await db.collection('bundle_metadata').ensureIndex({bundleId : 1}, {unique: true});
+
+  const extractMetadata = (bundle) => {
+    const {metadata} = bundle;
+    const {bundleId} = bundle;
+    const result = {bundleId, ...metadata};
+    return result;
+  };
+
+  const performMetadataMigration = async (bundle) => {
+    const metadata = extractMetadata(bundle);
+    try {
+      await db.collection('bundle_metadata').insertOne(metadata);
+    } finally {
+      await db.collection('bundles').updateOne({bundleId: bundle.bundleId}, {$unset: {metadata: 1}});
```

Comment:
There's a small problem with it: if db.collection('bundle_metadata').insertOne(metadata) fails for some reason, we later delete metadata here and it's gone FOREVER

Reply:
I agree with @vanruch on this. By doing this update inside of finally, we can potentially loose data if the insert fails.
If the try is here to catch a unique index conflict, we should instead do a `if` with `findOne`. If any of the actions then fails for 'really unexpected reasons' the whole migration will fail, and the operator can debug and fix:D